---
title: "DE_analysis_eels"
author: "Ludovic Dutoit"
date: "3/11/2020"
output: github_document

---

First, I go through the dataset, here it is:
```{r, echo=T, message=F,warning=FALSE}
library(tidyr)
library(DESeq2)
library(pheatmap)
library(stringr)
library(ggplot2)
library(matrixStats)
library(pheatmap)
library(edgeR)
library("RColorBrewer")

counts=read.csv("results_files/RSEM_gene_counts.txt", sep="", head=T)
head(counts)
colnames(counts)



######### Do the filtering
samples=cbind(colnames(counts),rep(c("silver","yellow"),each=6))
rownames(samples)<-samples[,1]
samples=as.data.frame(samples[,-1])
colnames(samples)<-"treatment"
treatment <- factor(rep(c("silver","yellow"),each=6))
```

### Basic filtering
```{r}
des <- model.matrix(~treatment)
dge <- DGEList(counts=counts) 
keep <- filterByExpr(counts, des) # inside function of limma aware of the design to remove rad counts
dge <- dge[keep,keep.lib.sizes=FALSE]



dge<- calcNormFactors(dge)
print("After filtering, kept a matrix of X genes by Y samples:")
dim(dge) 
```
### Some visualisation after basic filtering
```{r}

###all counts
logCounts = log2(as.matrix(counts)+ 0.5)
boxplot(as.matrix(logCounts) ~ col(logCounts),names=c("s1","s2","s3","s4","s5","s6","y1","y2","y3","y4","y5","y6"),main= "log Raw counts",sub="Y has a lot of transcripts with low counts",main="Counts before filtering")

###dge
logCounts = log2(as.matrix(dge)+ 0.5)
boxplot(as.matrix(logCounts) ~ col(logCounts),names=c("s1","s2","s3","s4","s5","s6","y1","y2","y3","y4","y5","y6"),main= "log filtered counts")
print(paste(c("number of transcripts:"," before filtering ",dim(counts)[1]," after filtering ",dim(dge)[1])))

print("While we get rid of most transcripts (lowly expressed), we keep a high proportion of all reads per samples")
apply(dge,2,sum)/apply(counts,2,sum)
```
### Basic stats post filtering

How many reads per gene? This is the distribution in 5% quantiles of number of reads per sample per gene
```{r}
quantile(apply(dge$counts,1,sum)/12,seq(0,1,0.05))
```
Up to 1e6 (maybe rRNA, to check). It means the most expressed genes has 1mio/reads per sample

median is about ten.

Let's visualise it for the bottom 85% (up to 2000)


```{r}
hist(apply(dge$counts,1,sum)/12,xlim=c(0,2000),breaks=100000,main="number of reads per gene per sample",xlab= "number of reads per sample",ylab= "frequency") # 32447 genes passing filtering
```

What are the most expressed genes above 1000000 per sample:


```{r}
which(apply(dge$counts,1,sum)/12>1e6)
```

TRINITY_DN2028_c0_g1 sequence matches Anguilla japonica eZPCa mRNA for egg envelope protein, complete cds:

```
>TRINITY_DN2028_c0_g1_i11 RANDOMLY PICKED ISOFORM
TACAGTAACCTCATGATCCATGCAGTCTGGCTACTGCATTGAAAGCTGCTTGGGGGGGGGGGGGAAATCCAAAACTTGATATAATGGAAGGTACAGAAATGTGGACTCAGAACACAATGATCTGGACAGCACGTTTTGAGGCACTTCAAGTTTTATTCAAAAATCCCCACCCAGTTACAGGACAGGTTTGTAGCGTCTCCACATCAGCACCATCCCCAGCGCAATGATGCAGACCAGTCCCACTGCAGCCATCACTCCAGCCATAAATACCACCTTAGAGGCACCTTCTGCTTGGTGATCTGCATTCAGAGGGCTGATGGACTCAGGTACATCTTTGACAGCCTCCTGGACAACAATTGGGCCTAGGACTGCGCCACCCTCATACTGTATGCCTGTATTCACACTCCTTATCTTCCTCCCACCACAGCTGGTGTCACAGCAGCTACATGCCTGGTCATTCCCAGATGCCTCAATCCACCGTTCCTTCCCAAGAGGAAATGAGCAAGCCTTCCCTTCTGAATCTGGCAAAGCTGCAGTAGCCTTCAGATAGCAGAAAATGTAGACAGCACTCCTGGTCTCCTGGGCAAACCTGAAGGCATCCAGCTGAAGTTGCAGCTTGTCGTCCTGTACCCTTGACAGGAACTGGGAGCGGGAGTTGGTCAGCTTGGAATCCATTAGACACCCCTTGTTCTCAATGAAAGCATATCTGGGGACAGCGTTCATGTCAGGGTCCAAAGTAGCTATACAGGTATCCACAAACACGCGGAGGGGCACGTGGTTGGCCTGGACCACTGAAGCCTCAATGTTGATGAGGTCACCCAGGAAGAACACCGTGGATGTCCTCTCCGTCTGCCAGTTATCAGCCATGAGCCTTAATGAAAATACAAGGAGGTCCTCAGCAGAAAGGGTGGAGTGGTAGGGGATCCAGGTTGGCTTCAGGGCATTACTGCTCACGTTATGCAGCCTCATGTAATGACACTGGATGCCCACCACTGCACTTGATGTCCTAATGATGGGAGTGGCACCAAGTGCTTTAGGTTGGTAGTTCAGGGTGAAGGTGTACACTAGGGAGTCAGCAGTCATTGCAAGCACACTGCCACAGCTGTGCAGCTCAGTCTCAAATAGAAGCACTTGGGTAGACCCATCCAGGTCAACAGGTCCGCAGCCTCCTAGGGTGATGTCAGTGGGCTGGATAAGGTGGCCGATGCCAAGCAGGTCCATATCTACTTCCAACTGAACAGAGCTTTCCCCACAGTGAGCCTTCACAGCATCTGGCTTAGGTGTAGGAGGTTCCACTGGAATTGTTGACC
```
TRINITY_DN4587 matches  PREDICTED: Anguilla anguilla zona pellucida sperm-binding protein 4-like (LOC118223083), mRNA

```
>TRINITY_DN4587_c0_g1_i12 len=1019 path=[4:0-44 6:45-214 7:215-228 8:229-252 9:253-342 11:343-487 12:488-566 14:567-597 15:598-681 17:682-711 19:712-968 20:969-1018]
ACTTCACTGTGAAGCCTCTGGTCGGTGGCTGTAAGCTCCGTATTGGTCAGGACTACAGGCTTACTGGAAACCACAGCCTTTTCCTCAGGGGACCGCTTGCCTACAGGAGCAATTGACCTTCTTTGCTTTCCACTGCTGCACCGCTGTTCACAGCGATCGGTAGCAGAGGGGTGGCAAACTGCCGTGCTACAGTGGATGAACACCGTTTCCTTCAGTGGAACGAAGGATTCAGGATCCACAAAGGTAAACATCTGCACAATAAACCGCTTGTAGTGACTGGGGAAGAGAAGTCCAGAAGAGGCATCTACAGGAACCAAGGATGTCTGGTAGCGGTCATCACGGTAGGGAGACCCATCAACCAAAAGGCTCCACTGGGGTAGGCTGAGGGGGCTGGAGGTTGATGTAGCCCAGCAGTGTTCCAGAAGCAGGACAAGGTTTGGGTCAGTCCTCTCCAAGATCCGCACTTCCACATACACAGGTTCCCGTAGCACCTTGGTCACGGGGTAGTCCGCATCTCCGTAGTAGTCGCTGTACACAGCATCAGCACATCCACTTGCTCCTTTGGAGTCACATTGACCACTAGCCAGTCTGAGCTCAATGCGAAGGGGCCCTGGAGCAGCTACTGGAAGGGGAGGCGGCACTGTATTCACCTCAGCTACTAAAGAAACCACAGCACTGCCAGAATACCTGCACTGGAAGGACAGCTCATAAATACTGTCCCTTGTGATGGAACCTAAAGGGCCAACGCCCACTTCATATGTGGAGGACATCTTGTTCTCATAGATCACGTAATCCCCTTCCACCCTCACTGTGGTTCCACAGTCACTGACTGGAAATTGGAAGACCGCAAAAGCAGGAGAGGCACTAACAGGACCACAGTAACCGCCACTGGGGCCCTCCAGCAAGTTCACAGAAGGAAGGTCAAGCAGAGGCAAGGTGGCGGCTCTGGACACTACAACCATGAACTGGCCATCCCGGAGACAGTGGACAGTAACTTCATTTGCATAATAGCACTGTTG
```


### DE analysis 

```{r}
logCPM <- cpm(dge, log=TRUE)
design <- model.matrix(~samples$treatment)

v <- voom(dge, design, plot=TRUE, normalize="quantile")
plotMDS(v, labels=c("s1","s2","s3","s4","s5","s6","y1","y2","y3","y4","y5","y6"),main="MDS plot", col=rep(c("gray","gold"),each=6))



fit <- lmFit(v, design)
fit <- eBayes(fit, trend=TRUE)
results<-topTable(fit, coef=ncol(design),n=Inf)

sum(results$adj.P.Val<0.05)
##Table of DE genes

#DE_counts are the counts for the DE 
DE_results<-results[results$adj.P.Val<0.05,] # DE_results are only the significant genes(p-value, log Fold-change,,,)
 
DE_counts<-dge[which(rownames(dge)%in%rownames(results)),]$counts #counts for significant genes


#positive means more in yellow
#negative means more in gray


##Overexpressed and underexpressed in general?
length(which(DE_results$logFC<(-1))) # how many genes under expressed in yellow  or more
length(which(DE_results$logFC>1)) # how many genes over expressed in yellow



##Overexpressed and underexpressed  by a actor 2 or more
length(which(DE_results$logFC<(-1))) # how many genes under expressed in yellow by a 
length(which(DE_results$logFC>1)) # how many genes over expressed in yellow


###Volcano plot (relationship between Log Fold Change and -value)
plot(results$logFC,-log(results$adj.P.Val),pch=10,cex=0.1) 
points(DE_results$logFC,-log(DE_results$adj.P.Val),pch=10,cex=0.1,col="red")



plot(results$AveExpr,results$logFC,pch=10,cex=0.1)
points(DE_results$AveExpr,DE_results$logFC,pch=10,cex=0.1,col="red")
```

Let's output those results as text tables that everyone can play with ( in excel or other). I  saved it as [DE_results.txt](DE_results.txt)


```{r}
final_results_DE<-merge(DE_results,DE_counts,by=0)
colnames(final_results_DE)[1]<-"GeneID"
write.table(final_results_DE,"DE_results.txt",row.names=T,col.names=T,sep="\t")
```

we also output the 32447 genes representing the whole genome for the GOSeq analysis:

```{r}
write(rownames(dge$counts),"backgroundGO.txt",sep="\n")
write(rownames(dge$counts),"backgroundALLGO.txt",sep="\n")

```

And finally we upload the results for ALL the genes:

```{r}
final_results<-merge(results,dge$counts,by=0)
colnames(final_results)[1]<-"GeneID"
write.table(final_results,"ALLgenes_results.txt",row.names=T,col.names=T,sep="\t")
```


### Saving data for plotting_genes.R

```{r}
###Careful, dirty fix 
counts<-cpm(dge$counts)
save(counts,results,file="countsand_logFCforplottingGene.RData")
rm(counts) #to make sure I don't use it thinking it is counts
```
